File: BJ/9012_괄호.py
T = int(input())


def check(strings):
    stack = []

    for i in strings:
        if i == '(':
            stack.append(i)
        else:
            if not stack:
                print("NO")
                return
            else:
                stack.pop()

    if not stack:
        print("YES")
        return
    else:
        print("NO")
        return
첫째 줄에 (A+B)%C, 둘째 줄에 ((A%C) + (B%C))%C, 셋째 줄에 (A×B)%C, 넷째 줄에 ((A%C) × (B%C))%C를 출력한다.

for _ in range(T):
    n = input()
    check(n)

File: Quiz/BJ/9012_괄호.cpp
#include <iostream>
using namespace std;
#define MAX 50

int main(){
	int T;
	cin>>T;
	
	string stack;
	const char* p = stack.c_str();
	int top = -1;
	
	while(T--) {
		cin>>stack;
		
		for(p = stack.c_str(); *p!='\0'; p++){
			if(*p == '('){			// push
				if(top == stack.length()){
					cout<<"NO"<<endl;
					goto end;
				}else{
					top = top + 1;
				}
			}else{					// pop
				if(top == -1){
					cout<<"NO"<<endl;
					goto end;
				}else{
					top = top - 1;
				}
			}
		}
		if(top == -1)
			cout<<"YES"<<endl;
		else{
			cout<<"NO"<<endl;
		}
		
	end:
		stack = "";
		top = -1;
	}
	
	return 0;
}
